compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -185,12 +185,18 @@ object Checking {
   /** Check that `tp` refers to a nonAbstract class
    *  and that the instance conforms to the self type of the created class.
    */
-  def checkInstantiable(tp: Type, pos: SrcPos)(using Context): Unit =
+  def checkInstantiable(tp: Type, srcTp: Type, pos: SrcPos)(using Context): Unit =
     tp.underlyingClassRef(refinementOK = false) match
       case tref: TypeRef =>
         val cls = tref.symbol
-        if (cls.isOneOf(AbstractOrTrait))
-          report.error(CantInstantiateAbstractClassOrTrait(cls, isTrait = cls.is(Trait)), pos)
+        if (cls.isOneOf(AbstractOrTrait)) {
+          srcTp.underlyingClassRef(refinementOK = false) match 
```

Comment:
It seems you could use `srcTp.underlyingClassRef(refinementOK = false).typeSymbol.is(Trait)` .

Reply:
Yes. Thanks.
With `srcTp.underlyingClassRef(refinementOK = false).typeSymbol` I can get rid of pattern matching and seems all tests still pass.

Please, take a look at pushed changes.